s2wav/Dimension helper functions.py

```diff
@@ -0,0 +1,373 @@
+def n_lm_scal(upsample: bool, L: int, kernel: str, J_min: int, lam: float) -> int:
+    '''Computes the square of the band-limit, after determining what the value of the band-limit is.
+
+    Args:
+        upsample (bool): Boolean parameter which determines whether to store the scales at j_max resolution or its own resolution.
+        L (int): Upper harmonic band-limit.
+        kernel (str): The wavelet type.
+        J_min (int): First wavelet scale to be used.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+    
+    Returns:
+        The square of the band-limit (int).
+    '''
+    if bandlimit == upsample:
+        bandlimit = L
+    else:
+        bandlimit = min(bandlimit(kernel, J_min - 1, lam, L), L)
+
+    return bandlimit * bandlimit
+
+
+def n_lmn_wav(lam: float, L: int, J_min: int, upsample: bool, kernel: str, N: int, storage: str, reality: int) -> int:
+    '''Computes the flmn sampling size.
+
+    Calls upon functions originally defined in so3, which have been defined explicitly for the case needed here.
+    
+    Args:
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        L (int): Upper harmonic band-limit.
+        J_min (int): First wavelet to be used.
+        upsample (bool): Boolean parameter which determines whether to store the scales at j_max resolution or its own resolution.
+        kernel (str): The wavelet type.
+        N (int): Upper azimuthal band-limit.
+        storage (str): The type of storage (padded or compact).
+        reality (int): A non-zero value indicates the signal, f, is real.
+ 
+    Returns:
+        flmn_size (int).
+    '''
+
+    J = j_max(lam, L)
+
+    total = 0
+
+    for i in range(J_min, J):
+        if not upsample:
+            bandlimit = min(bandlimit(kernel, i, lam, L), L)
+            L = bandlimit
+            N = min(N, bandlimit)
+
+        if storage.lower() == "padded":
+            if reality != 0:
+                sampling_flmn_size = N*L*L
+            else:
+                sampling_flmn_size = (2*N-1)*L*L
+        elif storage.lower() == "compact":
+            # Both of these are based on the face that the sum over n*n from 1 to N-1 is (N-1)*N*(2N-1)/6.
+            if reality != 0:
+                sampling_flmn_size = N*(6*L*L-(N-1)*(2*N-1))/6
+            else:
+                sampling_flmn_size = (2*N-1)*(3*L*L-N*(N-1))/3
+        else:
+            print("Invalid storage method.")
+
+        total += sampling_flmn_size
+
+    return total
+
+
+def n_gamma(N: int, steerable: int) -> int:
+    '''Computes the number of gamma samples for a given sampling scheme
+    
+    Args:
+        N (int): Upper orientational band-limit. Only flmn with n < N will be stored.
+        steerable (int): A non-zero value indicates that the signal is steerable.
+
+    Returns:
+        ngamma (int): Number of gamma samples.
+    '''
+    if steerable != 0:
```

Comment:
Steerable can be a boolean (I know in the C code it is an int but this is another case where C was just using ints as proxy for boolean).

Either way here, it's best to do 

`if steerable:`

which will check if steerable == 1 (i.e. is steerable true), rather than asking is steerable not equal to False which is a double negative :)

Reply:
Should I change steerable to be a boolean type then?

Reply:
@JessWhitney yeah that would be best I think.